integrations/paddleocr/src/haystack_integrations/components/converters/paddleocr/paddleocr_vl_document_converter.py

```diff
@@ -0,0 +1,491 @@
+# SPDX-FileCopyrightText: 2025-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+import base64
+from pathlib import Path
+from typing import Any, Literal, Optional, Union
+
+import requests
+from haystack import Document, component, default_from_dict, default_to_dict, logging
+from haystack.components.converters.utils import (
+    get_bytestream_from_source,
+    normalize_metadata,
+)
+from haystack.dataclasses import ByteStream
+from haystack.utils import Secret, deserialize_secrets_inplace
+from paddlex.inference.serving.schemas.paddleocr_vl import InferRequest as PaddleOCRVLInferRequest  # type: ignore
+from paddlex.inference.serving.schemas.paddleocr_vl import InferResult as PaddleOCRVLInferResult  # type: ignore
+from paddlex.inference.serving.schemas.shared.ocr import FileType  # type: ignore
+from typing_extensions import Self, TypeAlias
+
+logger = logging.getLogger(__name__)
+
+
+FileTypeInput: TypeAlias = Union[Literal["pdf", "image", 0, 1], None]
```

Comment:
Since this is meant to be provided by the user and later converted to `FileType`, can we just restrict it to
`FileTypeInput: TypeAlias = Union[Literal["pdf", "image"], None]`?